### DISCUSSION TITLE
PositionZ for .ctb files

### DISCUSSION THREAD

**Discussion by 98mcveigh** [sn4k3/UVtools#718] [Scripts]
I am trying to write a script that edits the Position Z parameter for .ctb files but it throws an error when I try to access PositionZ. However, I can see that PositionZ is a parameter in FileFormats/Chitubox.cs.

I am using Visual Studio to write the script and the example ScriptSetLiftHeightSample.cs worked fine. I did not get the pop-up to install the extension (not sure what extension it is/if I can install it some other way?) but the default script worked okay so I did not give it too much thought.

![error](https://github.com/sn4k3/UVtools/assets/38269180/2856cb86-c2a1-4686-9d2b-34c657017aa8)

**Comment by sn4k3** [comment]
❌ `PositionZ` is not member of `FileFormat`.
✅ `PositionZ` is member of `Layer`

You can access each layer PositionZ by issue the layer index: `SlicerFile[index].PositionZ = myvalue;`
